### PR TITLE
Add data source for oncall integration

### PR DESCRIPTION
**Pull request by karl-johan-grahn**
https://github.com/stakater/grafana-cloud-ansible-operator creates and managed oncall integrations, however integrations don't exist as data sources in the provider so cannot be referenced as such when creating oncall routes for example

### CODE REVIEW

File: internal/resources/oncall/data_source_integrations_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+				Config:      testAccDataSourceIntegrationConfig(integrationID),
+				ExpectError: regexp.MustCompile(`couldn't find an integration`),
+			},
+		},
+	})
+}
+
+func testAccDataSourceIntegrationConfig(integrationID string) string {
+	return fmt.Sprintf(`
+data "grafana_oncall_integration" "test-acc-integration" {
```
Comment (julienduchesne) [inline internal/resources/oncall/data_source_integrations_test.go]: I think the test should create an integration and then fetch it back

File: examples/resources/grafana_oncall_route/resource.tf [hunk 2]
```diff
@@ -2,13 +2,19 @@ data "grafana_oncall_slack_channel" "example_slack_channel" {
   name = "example_slack_channel"
 }
 
+data "grafana_oncall_integration" "example_integration_data_source" {
+  id = "CABCDEFGHI321"
+}
+
```
Comment (julienduchesne) [inline examples/resources/grafana_oncall_route/resource.tf]: hmm, I'd remove this one. The link between integration and route is already in this example file

File: internal/resources/oncall/data_source_integrations.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		ReadContext: withClient[schema.ReadContextFunc](dataSourceIntegrationRead),
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The integration ID.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
```
Comment (julienduchesne) [inline internal/resources/oncall/data_source_integrations.go]: It'd be nice if that could be specified as well. Input just the name -> get the ID and vice-versa

File: internal/resources/oncall/data_source_integrations.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package oncall
```
Comment (julienduchesne) [inline internal/resources/oncall/data_source_integrations.go]: This file should be named `data_source_integration.go`

File: examples/resources/grafana_oncall_route/resource.tf (hunk 2 above)
Comment (karl-johan-grahn) [reply examples/resources/grafana_oncall_route/resource.tf]: Alright removed it

File: internal/resources/oncall/data_source_integrations.go (hunk 4 above)
Comment (karl-johan-grahn) [reply internal/resources/oncall/data_source_integrations.go]: Renamed them

File: internal/resources/oncall/data_source_integrations.go (hunk 3 above)
Comment (karl-johan-grahn) [reply internal/resources/oncall/data_source_integrations.go]: The API https://grafana.com/docs/oncall/latest/oncall-api-reference/integrations/#get-integration just accepts an ID so I don't know how we would go about with that, would need advice if that would be a strict requirement

File: internal/resources/oncall/data_source_integrations_test.go (hunk 1 above)
Comment (karl-johan-grahn) [reply internal/resources/oncall/data_source_integrations_test.go]: I've made an attempt

**Comment by karl-johan-grahn** [comment]
Please run tests and review the tests

**Comment by karl-johan-grahn** [comment]
What about the docs, should I generate them and commit the generated files? Not clear if the pipeline will do that or if it's on me

**Comment by julienduchesne** [comment]
Merged through https://github.com/grafana/terraform-provider-grafana/pull/1532. Thank you!

**Pull request closed without merging**